pact.cabal

```diff
@@ -47,7 +47,7 @@ flag tests-in-lib
   manual:      True
 
 library
-
+  cpp-options: -DLEGACY_PARSER=1
```

Comment:
A couple thoughts

You don't need to set `=1` for binary flags, you can just define `-DLEGACY_PARSER`. 
Then CPP if-blocks become like
```haskell
#ifdef LEGACY_PARSER
...
#else
...
#endif
```
or
```haskell
#if someCondition
...
#elifdef LEGACY_PARSER
...
#endif
```

The other thought is that it could be behind a flag:
```cabal
flag legacy-parser
  description: ...
  default: True
  manual: True
  
library
  if flag(legacy-parser)
    cpp-options: -DLEGACY_PARSER
```
Which would make toggling it on/off not require editing the cabal file.

Reply:
Agreed, this is all temporary for now. I tried to define the macro in `cabal.project.local` but cabal doesn't like that, so I had to put it into the cabal file.

Regarding using 1/0, sometimes I like to be more explicit when dealing with macros (I find it easy to mess up with `#define` and `#undef`). Yeah, if that macro would still be in the final version of the code, we would use a standard pattern with (automatic) cabal flag. But, hopefully, that won't be necessary.